### PR TITLE
drivers: flash: flash_mcux_flexspi_nor: fix IS25WP flash support

### PR DESCRIPTION
**Pull request by danieldegrasse**
Some NXP boards program the IS25WP flash read parameters bits (P[6:3]) to use a higher frequency at boot, which interferes with the SFDP probe commands in the FlexSPI NOR driver. Reprogram these bits at boot time so the probe works as expected.

This PR also corrects the following issues with the FlexSPI NOR driver:
- Write enable instruction should be issued before setting the status register for quad enable
- After setting quad enable, the flash will be busy. The busy bit should be polled until it is ready to be accessed again
- Quad enable method 5 reads from status register 2, and writes to status register 1. We need to shift the buffer value written to account for this difference
- move the LUT table used during probing to the `.data` section, to reduce stack usage
- The maximum frequency for the IS25WP flash when using SFDP probe is 104 MHz. Set this correctly for boards using the flash chip.

Verified using `tests/drivers/flash/common` on the following boards:
- mimxrt1020_evk
- mimxrt1060_evk
- mimxrt1064_evk
- mixmrt1170_evk//cm7

### CODE REVIEW

**Comment by danieldegrasse** [comment]
Rebased to pick up CI fixes

File: drivers/flash/flash_mcux_flexspi_nor.c [hunk 1]
```diff
@@ -886,6 +908,52 @@ static int flash_flexspi_nor_check_jedec(struct flash_flexspi_nor_data *data,
 
 	/* Switch on manufacturer and vendor ID */
 	switch (vendor_id & 0xFFFF) {
+	case 0x609d: /* IS25LP flash, needs P[4:3] cleared with same method as IS25WP */
+		read_params = 0xE0U;
```
Comment (pdgendt) [inline drivers/flash/flash_mcux_flexspi_nor.c:912]: Add `__fallthrough`? But the `read_params` is reset on line 919.

File: drivers/flash/flash_mcux_flexspi_nor.c [hunk 2]
```diff
@@ -886,6 +908,52 @@ static int flash_flexspi_nor_check_jedec(struct flash_flexspi_nor_data *data,
...
+				kFLEXSPI_Command_WRITE_SDR, kFLEXSPI_1PAD, 0x1);
+		ret = memc_flexspi_set_device_config(&data->controller,
+					&config,
+					(uint32_t *)flexspi_lut,
+					FLEXSPI_INSTR_END * MEMC_FLEXSPI_CMD_PER_SEQ,
+					data->port);
+		if (ret < 0) {
+			while (1) {
+				/* Spin here, this flash won't configure correctly */
+			}
```
Comment (pdgendt) [inline drivers/flash/flash_mcux_flexspi_nor.c:947]: Print an error, a panic/oops...?

File: drivers/flash/flash_mcux_flexspi_nor.c (hunk 1 above)
Comment (danieldegrasse) [reply drivers/flash/flash_mcux_flexspi_nor.c:949]: I don't know how I missed this- thanks for catching it. I guess the IS25LP flash will function correctly with P[7:0] set to 0x0, but it is not correct for what we are trying to change here. Updated this section to use a helper function for both flashes.

File: drivers/flash/flash_mcux_flexspi_nor.c [hunk 3]
```diff
@@ -585,11 +591,26 @@ static int flash_flexspi_nor_quad_enable(struct flash_flexspi_nor_data *data,
...
+	if (qer == JESD216_DW15_QER_VAL_S2B1v5) {
+		/* Left shift buffer by a byte */
+		buffer = buffer << 8;
+	}
 	buffer |= bit;
 	transfer.dataSize = wr_size;
 	transfer.seqIndex = SCRATCH_CMD2;
 	transfer.cmdType = kFLEXSPI_Write;
-	return memc_flexspi_transfer(&data->controller, &transfer);
+	ret =  memc_flexspi_transfer(&data->controller, &transfer);
```
Comment (pdgendt) [inline drivers/flash/flash_mcux_flexspi_nor.c:607]: Nit, fix clang format notice.

**Review by pdgendt — DISMISSED** [review]

File: drivers/flash/flash_mcux_flexspi_nor.c (hunk 3 above)
Comment (danieldegrasse) [reply drivers/flash/flash_mcux_flexspi_nor.c:607]: Fixed, thanks

**Review by pdgendt — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by nashif** [comment]
@kartben  did your commit 8f4ee9bb813 break doc workflow status? Looks like we are now waiting for a check that will never happen?

**Comment by nashif** [comment]
> @kartben did your commit [8f4ee9b](https://github.com/zephyrproject-rtos/zephyr/commit/8f4ee9bb813bd697ccb726c3216d5945af70ced9) break doc workflow status? Looks like we are now waiting for a check that will never happen?

maybe I am wrong.

**Comment by kartben** [comment]
> > @kartben did your commit [8f4ee9b](https://github.com/zephyrproject-rtos/zephyr/commit/8f4ee9bb813bd697ccb726c3216d5945af70ced9) break doc workflow status? Looks like we are now waiting for a check that will never happen?
> 
> maybe I am wrong.

That's what I was thinking too but it doesn't make sense it would be happening only know, and the error message just looks as a (severe) GitHub glitch to me. Can someone please force push on this branch to retrigger CI, as the job seems so stuck it can't even be restarted?

**Comment by stephanosio** [comment]
> @nashif That's what I was thinking too but it doesn't make sense it would be happening only now, and the error message just looks as a GitHub glitch to me. Can someone please force push to this PR's branch to retrigger CI, as the docs job seems so stuck it can't even be restarted?

@kartben It seems to be a GitHub glitch indeed.

Overriding the "Documentation Build Status" check and merging since this PR does not contain any documentation changes.

**Comment by dleach02** [comment]
okay, probably should have let this stay so we can debug what is going on...

**Pull request merged**